### PR TITLE
Convert the desktop app into a server

### PR DESCRIPTION
**Pull request by CodeMooseUS**
This pull request started out as a learning exercise for me to figure out how switch homebrew works. I saw this app and thought it was great. The problem was, I kept having to go upstairs to my PC in order to connect back to the switch whenever I changed app. So I began looking into how to reverse the connection so that the desktop app acted as the server and the switch a client. 

Once I figured out how to build it, I added the steps to the readme (at least the ones I took- they might not be optimal). Then from there I updated the desktop app to listen for client connections so you can just keep it running all the time. After that I needed a way to enter my desktop app ip address from the switch. That led me to add SDL2 (which I saw you also recently added) and re-purpose some code from [NX-Shell](https://github.com/joel16/NX-Shell) to render an on screen keyboard (it uses the same license so I figured that was ok).

From there things just kept going because it was a fun project. I added an icon that I drew in inkscape and recently merged with your overclocking changes, etc.

There are still a couple of issues around closing the homebrew app, and disconnecting from the server, but I figured it was in a good enough state to see if you want to merge it and help fix the bugs. Sorry that it's so many changes in one go.

Features in this PR include:
- Desktop app now runs as a server so you don't need to enter the switch IP anymore
- Switch app now has an icon and updated UI
- Added build steps and updated the readme with sceenshots
- You can enter the desktop server IP directly in the switch app and connect
- You can set the overclock value from the UI
- ~~Settings are persisted in the switch app~~ Actually this is bugged right now
- Updated version number to 0.2

Screenshot of updated switch app UI:
switch

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -55,10 +67,24 @@ This app uses core overclocking to 1785 MHz on the Nintendo Switch. We use this
...
 
 ## Build instructions
-
-Use the PKGBUILD from [here](https://github.com/jakibaki/pacman-packages/tree/ffmpeg_networking/switch/ffmpeg) for ffmpeg on Switch with more protocols enabled.
-
-Everything else will follow here in short time (ask jakibaki on AtlasNX discord if necessary).
+The following steps were done on Windows 10.
+
+### Compile homebrew
+1. Install devkitpro from [here](https://devkitpro.org/wiki/devkitPro_pacman)
```
Comment (jakibaki) [inline Readme.md]: This seems to be windows-only. Please make the instructions more general.

File: source/context.h [hunk 2]
```diff
@@ -1,40 +1,20 @@
 #ifndef _CONTEXT_H
 #define _CONTEXT_H
 
-#include <libavformat/avformat.h>
 #include <switch.h>
-#include <SDL.h>
-#include <SDL2/SDL_image.h>
-#include "SDL_FontCache.h"
```
Comment (jakibaki) [inline source/context.h]: this results in a compiler-error on linux

`#include "sdl_fontcache.h"` seems to work

File: source/context.h [hunk 3]
```diff
@@ -1,40 +1,20 @@
...
 
 #define RESX 1280
 #define RESY 720
 
-
-typedef struct
-{
-    SDL_Window *window;
-    SDL_Renderer *renderer;
-    SDL_Texture *yuv_text;
```
Comment (jakibaki) [inline source/context.h]: Why are you changing the context so much?

File: source/main.c [hunk 4]
```diff
@@ -94,34 +137,217 @@ void startRender(VideoContext *videoContext)
     threadStart(&renderThread);
 }
 
-int main(int argc, char **argv)
+static void DrawTitle()
+{
+    SDL_ClearScreen(BACKGROUND_COLOR);
```
Comment (jakibaki) [inline source/main.c]: This shouldn't be in the main.c

File: source/main.c [hunk 5]
```diff
@@ -94,34 +137,217 @@ void startRender(VideoContext *videoContext)
...
+            if (!foundConnection)
+            {
+                printf("Connected to server\n");
+                foundConnection = true;
+            }
             displayFrame(renderContext);
-        } else {
-            drawSplash(renderContext);
+        }
+        else if (!foundConnection)
```
Comment (jakibaki) [inline source/main.c]: this shouldn't be in the main.c either

File: source/main.c [hunk 6]
```diff
@@ -94,34 +137,217 @@ void startRender(VideoContext *videoContext)
...
+	Touch_Init(&touchInfo);
+
+    int row = 0;
+
+    while(appletMainLoop()) 
+    {
+		DrawTitle();
+
+        // Server address textbox
+        SDLRect serverAddressRect = { 640 - 250, 350, 500, 80 };
```
Comment (jakibaki) [inline source/main.c]: shouldn't be in the main either

File: source/sdl_helpers.c [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+    sdl_helper: joel16/NX-Shell
+    See sdl_helpers.h for license info.
+
+    Modifications by CodeMooseUS
+*/
+
+#include "common.h"
+#include "sdl_helpers.h"
+
+static SDL_Window *WINDOW;
```
Comment (jakibaki) [inline source/sdl_helpers.c]: Why are you moving this here?

File: Windows/In-Home-Switching-Win/In-Home-Switching/Streamer.cs [hunk 8]
```diff
@@ -56,105 +30,164 @@ public struct InputPkg
...
+            Left = 1 << 12,
+            Up = 1 << 13,
             Right = 1 << 14,
-            Down  = 1 << 15
+            Down = 1 << 15
         }
 
-        public void Start(string ip, int quality)
+        // Members
+        private const string FFMPEGArgs = "-y -f rawvideo -pixel_format rgb32 -framerate 300 -video_size {0}x{1} -i pipe: -f h264 -vf scale=1280x720 -preset ultrafast -tune zerolatency -pix_fmt yuv420p -profile:v baseline -x264-params \"nal-hrd=cbr\" -b:v {2}M -minrate {2}M -maxrate {2}M -bufsize 2M tcp://{3}:2222 ";
```
Comment (jakibaki) [inline Windows/In-Home-Switching-Win/In-Home-Switching/Streamer.cs]: why is the ffmpeg not running as server?

**Comment by CodeMooseUS** [comment]
That's fair. I realised I had done too much in one go, but it was a fun learning experience for me.

I'll abandon this PR for now, but feel free to use any pieces from the branch that look interesting.

Thanks.

**Pull request closed without merging**